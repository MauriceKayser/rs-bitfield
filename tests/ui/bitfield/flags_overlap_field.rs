//! Assertion failure: _FLAGS_IN_FIELD_1_OVERLAP_WITH_FIELD_0

extern crate alloc;

#[bitfield::bitfield(8)]
struct BitField {
    #[field(0, 2)] // This field overlaps with the flags F1, F2 & F3.
    field: Field,
    flags: Flags
}

#[derive(Debug, bitfield::Field)]
#[repr(u8)]
enum Field {
    F1 = 1,
    F2,
    F3
}

#[derive(Copy, Clone, Debug, bitfield::Flags)]
#[repr(u8)]
enum Flags {
    F0,
    F1,
    F3 = 3
}

fn main() {}