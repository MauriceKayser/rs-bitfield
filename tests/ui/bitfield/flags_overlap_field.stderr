error[E0080]: evaluation of constant value failed
 --> tests/ui/bitfield/flags_overlap_field.rs:8:12
  |
8 |       field: Field,
  |  ____________^
9 | |     flags: Flags
  | |________________^ the evaluated program panicked at 'Flags overlap with field', $DIR/tests/ui/bitfield/flags_overlap_field.rs:8:12
  |
  = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `panic` (in Nightly builds, run with -Z macro-backtrace for more info)
